packages/matter.js/src/device/Device.ts

```diff
@@ -33,7 +33,15 @@ export class PairedDevice extends Endpoint {
         clusters: (ClusterServerObj<Attributes, Commands> | ClusterClientObj<Attributes, Commands>)[] = [],
         endpointId: number
     ) {
-        super(definition, clusters, endpointId);
+        super(definition, endpointId);
+        clusters.forEach(cluster => {
+            if (isClusterServer(cluster)) {
```

Comment:
Just an aside but might be cleaner to have "isClient = true; isServer = false;" on the client and inverse on server.  Then TS will use as a discriminator so no need to import is functions

Reply:
In fact the functions are type hints, but yes additional such attributes would also have worked